File: ABC118/B.py
N, M = map(int, input().split())
Fm = [0]*M
KA = [list(map(int, input().split())) for _ in range(N)]
for n in range(N):
  #KA = list(map(int, input().split()))
  for i in range(KA[n][0]):
    Fm[KA[n][i+1]-1] += 1
ans = 0
for m in range(M):
  if Fm[m] == N:
    ans += 1
print(ans)

File: ABC118/B.cpp
#include<stdio.h>
  
int main(){
  int N, M, K, A, Fm[30] = {0}, i, j, ans = 0;
  
  scanf("%d %d", &N, &M);
  
  for (i = 0; i < N; i++){
    scanf("%d", &K);
    for (j = 0; j < K; j++){
      scanf("%d", &A);
      Fm[A - 1]++;
    }
  }
  for (i = 0; i < M; i++){
    if(Fm[i] == N){
      ans++;
    }
  }
  
  printf("%d\n", ans);
  
  return 0;
}